nextcord/gateway.py

```diff
@@ -721,17 +752,19 @@ class DiscordVoiceWebSocket:
     CLIENT_CONNECT      = 12
     CLIENT_DISCONNECT   = 13
 
+    if TYPE_CHECKING:
+        _connection: VoiceClient
+        gateway: str
+        _max_heartbeat_timeout: float
+        thread_id: int
+
     def __init__(self, socket, loop, *, hook=None):
         self.ws = socket
         self.loop = loop
-        self._keep_alive = None
+        self._keep_alive: Any = None
         self._close_code = None
         self.secret_key = None
-        if hook:
-            self._hook = hook
-
-    async def _hook(self, *args):
-        pass
+        self._hook = hook or _hook
```

Comment:
Is this for sure equivalent?

One case I can see it being different is if you subclass `DiscordVoiceWebSocket` to override `_hook`, but I don't know if that's something that would normally be done since I haven't used this feature. (It seems to be mostly internal use?)

Reply:
im barely sure on what this is either, it was a quickfix on solving a type issue with assigning to things like that, i could probably remake it to use getattr instead

Reply:
should be resolved i hope